images/macos/software-report/SoftwareReport.Xamarin.psm1

```diff
@@ -13,66 +10,26 @@ function Get-NUnitVersion {
     return "NUnit ${version}"
 }
 
-function Build-MonoList {
-    $monoVersionsPath = "/Library/Frameworks/Mono.framework/Versions"
-    $monoFolders = Get-ChildItemWithoutSymlinks $monoVersionsPath
-
-    $monoVersionList = $monoFolders | ForEach-Object {
-        $versionFilePath = Join-Path $_.FullName "Version"
-        if (Test-Path $versionFilePath) {
-            return Get-Content -Raw $versionFilePath
-        }
-
-        return $_.Name
-    } | ForEach-Object { $_.Trim() }
-
-    return $monoVersionList | Sort-Object -Property $sortRulesByVersion
-}
-
-function Build-XamarinIOSList {
-    $sdkVersionsPath = "/Library/Frameworks/Xamarin.iOS.framework/Versions"
-    $sdkFolders = Get-ChildItemWithoutSymlinks $sdkVersionsPath
-
-    $sdkVersionList = $sdkFolders | ForEach-Object {
-        $versionFilePath = Join-Path $_.FullName "Version"
-        if (Test-Path $versionFilePath) {
-            return Get-Content -Raw $versionFilePath
-        }
-
-        return $_.Name
-    } | ForEach-Object { $_.Trim() }
-
-    return $sdkVersionList | Sort-Object -Property $sortRulesByVersion
-}
-
-function Build-XamarinMacList {
-    $sdkVersionsPath = "/Library/Frameworks/Xamarin.Mac.framework/Versions"
-    $sdkFolders = Get-ChildItemWithoutSymlinks $sdkVersionsPath
-
-    $sdkVersionList = $sdkFolders | ForEach-Object {
-        $versionFilePath = Join-Path $_.FullName "Version"
-        if (Test-Path $versionFilePath) {
-            return Get-Content -Raw $versionFilePath
+function Build-XamarinTable {
+    $xamarinBundles = Get-ToolsetValue "xamarin.bundles"
+    $defaultSymlink = Get-ToolsetValue "xamarin.bundle-default"
+    if ($defaultSymlink -eq "latest") {
+        $defaultSymlink = $xamarinBundles[0].symlink
+    }
+    $sortRules = @{
```

Comment:
Why do we need sorting here? Can we just print bundle in the order how they are located in toolset.json?
Sorting by symlink as a string can be unreliable

Reply:
Removed this sort